addons/mail/models/mail_thread.py

```diff
@@ -2270,7 +2273,7 @@ def _notify_record_by_email(self, message, recipients_data, msg_vals=False,
                 if email_to:
                     create_values['email_to'] = email_to
                 create_values.update(base_mail_values)  # mail_message_id, mail_server_id, auto_delete, references, headers
-                email = Mail.create(create_values)
+                email = Mail.create(self._hook_mail_values(create_values))
```

Comment:
Why not a _notify_email_values() method encapsulating notify_email_headers ? Instead of

```
headers = self._notify_email_headers()
```

We could have

```
def _notify_email_values(base_mail_values):
    headers = self._notify_email_headers()
    if headers:
        base_mail_values['headers'] = headers
    return base_mail_values

... blahblah

base_mail_values = self._notify_email_values(base_mail_values)
```

Reply:
Done, thanks.